src/modules/ethereum/chains/requests.ts

```diff
@@ -22,46 +21,22 @@ function fetchChains(
         scope: ['info'],
         payload,
       },
-      onData: ({ value }) => {
+      onData: ({ value, isStale }) => {
         if (value) {
           resolve(value);
-          unsubscribe();
+          if (!isStale) {
+            unsubscribe();
+          }
         }
       },
     });
   });
 }
 
 export async function getNetworksBySearch({ query }: { query: string }) {
-  const networks = await fetchChains({
+  return fetchChains({
     include_testnets: true,
     supported_only: false,
     search_query: query.trim().toLowerCase(),
   });
```

Comment:
Let's add reject after delay here, too. It seems it's missing

Reply:
Agree!